docs/06-A-gdb-debugging-example.md

```diff
@@ -0,0 +1,52 @@
+## An example of how to debug WebKit GTK with gdb from a WebKit container
+
+1. Add a `.gdbinit` file as described at https://docs.webkit.org/Build%20%26%20Debug/DebuggingOnTheCommandLine.html#setting-up-your-environment.
+2. Add logging for the pid to the code intended to debug. E.g. to `SomeFile.cpp`:
+```
+#include <iostream>
+[...]
+SomeClass::SomeMethod() {
+std::cout << "pid=" << getpid() << std::endl;
+[...]
+}
+```
+3. Build WebKit from a container: `./Tools/Scripts/build-webkit --debug --gtk`
+4. Run MiniBrowser, from above container, with some file to debug, e.g.:
+```
+./Tools/Scripts/run-minibrowser --debug --gtk LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/lang-attribute-affects-rendering.html
+```
+6. Enter the same container in a different tab and (mentioned at https://docs.webkit.org/Build%20%26%20Debug/DebuggingOnTheCommandLine.html#manually-debugging-webkit):
+```
+export DYLD_FRAMEWORK_PATH=WebKitBuild/Debug
+```
+6. Find the logged `pid`, e.g. `pid=18102`, and attach gdb to that process:
```

Comment:
You can just look for WPEWebProcesses using `ps` -- why do you need to print the pid?